### ISSUE TITLE
still getting java.lang.NoClassDefFoundError with rest-assured:4.1.2

### ISSUE THREAD

**Issue by peter-ponzel** [rest-assured/rest-assured#1220]
hello,

according to 
https://github.com/rest-assured/rest-assured/issues/1168
https://github.com/rest-assured/rest-assured/issues/1175
this should have been fixed but sadly I'm still getting `java.lang.NoClassDefFoundError: io/restassured/path/json/mapper/factory/JohnzonObjectMapperFactory`
this exception is only thrown when I have the following piece of code in my test base
```
@BeforeClass
public static void setup() {
	RestAssured.port = 9001;
}
```
using
java 11
spring-boot:2.1.9
rest-assured:4.1.2

best regards,
peter

**Comment by makobernal** [comment]
I get the same exception on 4.1.2. In my case, it's only apparent when adding the following config to my test:
```
RestAssured.enableLoggingOfRequestAndResponseIfValidationFails()
```
I solved it importing additional modules:
```
    test "io.rest-assured:json-path:4.1.2"
    test "io.rest-assured:xml-path:4.1.2"
```
which is not ideal, since I'm not 100% sure I'd be using them, but it might become useful for someone.

**Comment by avierr** [comment]
Yeah same issue with JDK13 as well, had to revert to 4.1.1

**Comment by avierr** [comment]
ok I was was wrong, fix for this is to import both:

	testCompile group: 'io.rest-assured', name: 'rest-assured', version: '4.1.2'
	testCompile group: 'io.rest-assured', name: 'rest-assured-all', version: '4.1.2'

I hope there is a proper fix for this

**Comment by ETLJ** [comment]
> ok I was was wrong, fix for this is to import both:
> 
> ```
> testCompile group: 'io.rest-assured', name: 'rest-assured', version: '4.1.2'
> testCompile group: 'io.rest-assured', name: 'rest-assured-all', version: '4.1.2'
> ```
> 
> I hope there is a proper fix for this

This worked for me too -- I was having the same issue.

**Comment by fabianMendez** [comment]
I had the same issue with `io.rest-assured:rest-assured:4.1.2` and `io.rest-assured:spring-mock-mvc:4.1.2`  but I am using `Java 8` with gradle 7.

Importing `rest-assured-all` fixed the problem too.

**Comment by borkke** [comment]
I had a same problem with Java 11 and rest assured 4.2.0.
I only had this dependency
```
testImplementation "io.rest-assured:rest-assured:${restAssuredVersion}"
```
After I added these two, everything worked fine
```
testImplementation "io.rest-assured:json-path:${restAssuredVersion}"
testImplementation "io.rest-assured:xml-path:${restAssuredVersion}"
```

**Comment by mamachanko** [comment]
With 
```
testImplementation('io.rest-assured:rest-assured:4.3.0')
```
I am getting 
```
java.lang.ClassNotFoundException: io.restassured.path.json.mapper.factory.JsonbObjectMapperFactory
```

Downgrading to `4.2.0` and include `-all` fixed it for me:
```
testImplementation('io.rest-assured:rest-assured:4.2.0')
testImplementation('io.rest-assured:rest-assured-all:4.2.0')
```

**Comment by vmauricio93** [comment]
@mamachanko answer worked for me in Java 11

**Comment by lchrennew** [comment]
> With
> 
> ```
> testImplementation('io.rest-assured:rest-assured:4.3.0')
> ```
> 
> I am getting
> 
> ```
> java.lang.ClassNotFoundException: io.restassured.path.json.mapper.factory.JsonbObjectMapperFactory
> ```
> 
> Downgrading to `4.2.0` and include `-all` fixed it for me:
> 
> ```
> testImplementation('io.rest-assured:rest-assured:4.2.0')
> testImplementation('io.rest-assured:rest-assured-all:4.2.0')
> ```

I came across the same problem with 4.3.3:

```
        <dependency>
            <groupId>io.rest-assured</groupId>
            <artifactId>rest-assured</artifactId>
            <version>4.3.3</version>
        </dependency>
```
And I downgrade to 4.2.0 and include -all, then it works

**Comment by ybekardev** [comment]
The same issue is resolved by switching back to the 4.1.1 version

<dependency>
			<groupId>io.rest-assured</groupId>
			<artifactId>rest-assured</artifactId>
			<version>4.1.1</version>
 		</dependency>
		<dependency>
			<groupId>io.rest-assured</groupId>
			<artifactId>rest-assured-common</artifactId>
			<version>4.1.1</version>
		</dependency>
		<dependency>
			<groupId>io.rest-assured</groupId>
			<artifactId>rest-assured-all</artifactId>
			<version>4.1.1</version>
		</dependency>

**Comment by peter-ponzel** [comment]
after some time I gave it another try with
```
@BeforeAll
public static void init() {
    RestAssured.port = 9001;
}
```
using
* spring-boot:2.6.3
* rest-assured:4.4.0
* both java 11 and 17

and now it seems to work .. ¯\_(ツ)_/¯
